### PR TITLE
Clarify use of Scheme

### PR DESCRIPTION
**Pull request by pmorie**
Attempts to clarify the use of `Scheme`, plus other miscellaneous godoc improvements.

### CODE REVIEW

**Comment by pmorie** [comment]
I think that's a bad merge - I'll fix it.

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -33,8 +33,8 @@ type Options struct {
 	// MaxConcurrentReconciles is the maximum number of concurrent Reconciles which can be run. Defaults to 1.
 	MaxConcurrentReconciles int
 
-	// Reconciler reconciles an object
-	Reconciler reconcile.Reconciler
+	// Reconcile is the reconciler for the type the controller manages.
+	Reconcile reconcile.Reconcile
```
[the remark this answers is not in the document]
Comment (DirectXMan12) [reply pkg/controller/controller.go]: we used to have one, but I think it got renamed *really* early to `Reconciler`.

**Comment by pmorie** [comment]
I haven't meant to let this rot - last couple weeks have been crazy for me. Will fix shortly.

**Comment by pwittrock** [comment]
@pmorie Still working on this?  @DirectXMan12 Maybe we can just fix it up directly through the GH UI and merge it?

**Comment by droot** [comment]
This PR -->  https://github.com/kubernetes-sigs/controller-runtime/pull/295 improved the docs. So closing this in favor of this.

**Pull request closed without merging**